### PR TITLE
Add wrapper for merging throttle and brake actions in Carla

### PR DESCRIPTION
**Pull request by Haichao-Zhang**
A wrapper for merging throttle and brake actions in Carla. 
It is convenient in cases such as visualization of the Q functions over the action space.

### CODE REVIEW

File: alf/environments/alf_wrappers.py [hunk 1]
```diff
@@ -831,6 +831,116 @@ def time_step_spec(self):
...
+        return self._time_step_spec
+
+    def _step(self, action):
+
+        unmerged_action = torch.zeros(*action.shape[0:-1],
+                                      self._full_action_dim)
+        # throttle
+        valid_mask_throttle = (action[..., 0] >=
+                               self._throttle_damping).float()
+        unmerged_action[..., 0] = valid_mask_throttle * action[..., 0]
```
Comment (emailweixu) [inline alf/environments/alf_wrappers.py]: I feel it is better to make the mapping continuous as `valid_mask_throttle * (action[..., 0] - self._throttle_dammping)`

File: alf/environments/alf_wrappers.py [hunk 2]
```diff
@@ -831,6 +831,116 @@ def time_step_spec(self):
...
+        # steer
+        unmerged_action[..., 1] = action[..., 1]
+        # brake
+        valid_mask_brake = (action[..., 0] <= -self._brake_damping).float()
+        unmerged_action[..., 2] = -(valid_mask_brake * action[..., 0])
```
Comment (emailweixu) [inline alf/environments/alf_wrappers.py]: same here

File: alf/environments/alf_wrappers.py [hunk 3]
```diff
@@ -831,6 +831,116 @@ def time_step_spec(self):
...
+        # reverse
+        if not self._exclude_reverse:
+            unmerged_action[..., 3] = torch.clamp(action[..., 2])
+
+        time_step = self._env._step(unmerged_action)
+
+        self._prev_action_merged = action
```
Comment (emailweixu) [inline alf/environments/alf_wrappers.py]: I think action is torch.Tensor and there is no need for to_tensor in the next line.

File: alf/environments/alf_wrappers.py [hunk 4]
```diff
@@ -831,6 +831,116 @@ def time_step_spec(self):
         return self._time_step_spec
 
 
+@alf.configurable
+class CarlaMergedActionWrapper(AlfEnvironmentBaseWrapper):
```
Comment (emailweixu) [inline alf/environments/alf_wrappers.py]: This is only useful for carla. It's better to put it in suite_carla

File: alf/environments/alf_wrappers.py (hunk 1 above)
Comment (Haichao-Zhang) [reply alf/environments/alf_wrappers.py]: Changed to a linear mapping to [0, 1].

File: alf/environments/alf_wrappers.py (hunk 2 above)
Comment (Haichao-Zhang) [reply alf/environments/alf_wrappers.py]: Change as well

File: alf/environments/alf_wrappers.py (hunk 3 above)
Comment (Haichao-Zhang) [reply alf/environments/alf_wrappers.py]: Removed related codes.

File: alf/environments/alf_wrappers.py (hunk 4 above)
Comment (Haichao-Zhang) [reply alf/environments/alf_wrappers.py]: Move this wrapper and also CarlaActionWrapper into carla_utils.

Comment (Haichao-Zhang) [reply alf/environments/alf_wrappers.py]: To make code review easier, I split the updates into to commits: 
1. Code changes: 76749c13ee385835788c7be52b6e4c26f9544dbf
2. Code movements: 83ee1c7fde3f57673d4d5462a8cd2e1ad1b3a886

**Review by emailweixu — APPROVED** [review]

**Pull request merged**